src/librustc/hir/print.rs

```diff
@@ -678,12 +678,14 @@ impl<'a> State<'a> {
             }
             hir::ItemImpl(unsafety,
                           polarity,
+                          defaultness,
                           ref generics,
                           ref opt_trait,
                           ref ty,
                           ref impl_items) => {
                 self.head("")?;
                 self.print_visibility(&item.vis)?;
+                self.print_defaultness(defaultness)?;
```

Comment:
I've changed the parser-lalr.y accordingly to the `[default] [unsafe] impl` order so this file should be ok now